internal/test/morerequire/morerequire.go

```diff
@@ -72,3 +72,25 @@ func RequireSetenv(t *testing.T, key, value string) func() {
 		require.NoError(t, err, `error reverting env variable %s=%s`, key, previous)
 	}
 }
+
+// RequireChdirIntoTemp creates a new temp directory and cleans it up on the returned function.
+func RequireChdirIntoTemp(t *testing.T) (cleanup func()) {
+	var cleanups []func()
+	cleanup = func() {
+		for i := len(cleanups) - 1; i >= 0; i-- {
+			cleanups[i]()
+		}
+	}
+
+	wd, err := os.Getwd()
+	require.NoError(t, err)
+	cleanups = append(cleanups, func() {
+		require.NoError(t, os.Chdir(wd))
+	})
+
+	tempDir, removeTempDir := RequireNewTempDir(t)
+	cleanups = append(cleanups, removeTempDir)
+
+	require.NoError(t, os.Chdir(tempDir))
+	return
+}
```

Comment:
```suggestion

func RequireChdirIntoTemp(t *testing.T) func() {
	wd, err := os.Getwd()
	require.NoError(t, err)
	tempDir, removeTempDir := RequireNewTempDir(t)
	return func() {
		require.NoError(t, os.Chdir(wd))
		removeTempDir()
		require.NoError(t, os.Chdir(tempDir))
	}
}
```

How about this?

Reply:
it forgot to chdir :D but sure can composite the func I think that's the thing you mainly want!

Reply:
added a commit similar!

Reply:
https://github.com/tetratelabs/getenvoy/commit/01331b8dd1bed331a273ec7b3a4c28a4d0fa8420